### ISSUE TITLE
FeatureFlagService failing to start Docker in Native Ubuntu and WSL 

### ISSUE THREAD

**Issue by julianocosta89** [open-telemetry/opentelemetry-demo#697]
# Bug Report

Which version of the demo you are using?
[[commit hash](https://github.com/open-telemetry/opentelemetry-demo/commits/main)](https://github.com/open-telemetry/opentelemetry-demo/commit/ddd0bc33f20d890425cc0ed60dd71c2d033c5ac3).

## Symptom

**What is the expected behavior?**

I'd like to see the FeatureFlagService starting up.

**What is the actual behavior?**

The FeatureFlagService repeatedly logs a couple of errors and never starts.

## Reproduce

In Windows WSL or in Ubuntu 22.04.1 LTS, simply run:
```
docker compose build --no-cache
docker compose up
```

We will close this issue if:

* The steps you provided are complex.
* If we can not reproduce the behavior you're reporting.

## Additional Context

I spent some time banging my head against the wall trying to pin point the issue, but the errors are not that clear to me.
I'm getting some weird logs like a bunch of numbers:
```
feature-flag-service     |                                  100,84,103,117,68,110,70,116,50,75,97,65,74,
feature-flag-service     |                                  74,105,70,113,89,103,73,119,99,100,75,49,
feature-flag-service     |                                  106,49,122,113,79,43,70,52,67,89,87,111,100,
feature-flag-service     |                                  90,73,55,10,121,70,122,57,83,79,56,78,100,
feature-flag-service     |                                  67,75,111,67,79,74,117,120,85,110,79,120,
feature-flag-service     |                                  119,121,56,112,50,70,112,56,102,99,55,52,83,
feature-flag-service     |                                  114,76,43,83,118,122,90,112,65,51,10,45,45,
feature-flag-service     |                                  45,45,45,69,78,68,32,67,69,82,84,73,70,73,
feature-flag-service     |                                  67,65,84,69,45,45,45,45,45,10,10,73,100,101
[...]
```
This appears over and over.

And after the numbers, this:
```
feature-flag-service  | ** When Server state == {state,false}
feature-flag-service  | ** Reason for termination ==
feature-flag-service  | ** {{badmatch,{error,enoent}},
feature-flag-service  |     [{pubkey_os_cacerts,get,0,[{file,"pubkey_os_cacerts.erl"},{line,38}]},
feature-flag-service  |      {tls_certificate_check_shared_state,
feature-flag-service  |          maybe_load_authorities_trusted_by_otp,3,
feature-flag-service  |          [{file,
feature-flag-service  |               "/app/deps/tls_certificate_check/src/tls_certificate_check_shared_state.erl"},
feature-flag-service  |           {line,362}]},
feature-flag-service  |      {tls_certificate_check_shared_state,new_shared_state,2,
feature-flag-service  |          [{file,
feature-flag-service  |               "/app/deps/tls_certificate_check/src/tls_certificate_check_shared_state.erl"},
feature-flag-service  |           {line,335}]},
feature-flag-service  |      {tls_certificate_check_shared_state,handle_shared_state_initialization,
feature-flag-service  |          2,
feature-flag-service  |          [{file,
feature-flag-service  |               "/app/deps/tls_certificate_check/src/tls_certificate_check_shared_state.erl"},
feature-flag-service  |           {line,262}]},
feature-flag-service  |      {gen_server,try_dispatch,4,[{file,"gen_server.erl"},{line,1123}]},
feature-flag-service  |      {gen_server,handle_msg,6,[{file,"gen_server.erl"},{line,1200}]},
feature-flag-service  |      {proc_lib,init_p_do_apply,3,[{file,"proc_lib.erl"},{line,240}]}]}
feature-flag-service  | 
feature-flag-service  | =CRASH REPORT==== 18-Jan-2023::21:41:02.440737 ===
feature-flag-service  |   crasher:
feature-flag-service  |     initial call: tls_certificate_check_shared_state:proc_lib_init/0
feature-flag-service  |     pid: <0.1748.0>
feature-flag-service  |     registered_name: tls_certificate_check_shared_state
feature-flag-service  |     exception error: no match of right hand side value {error,enoent}
feature-flag-service  |       in function  pubkey_os_cacerts:get/0 (pubkey_os_cacerts.erl, line 38)
feature-flag-service  |       in call from tls_certificate_check_shared_state:maybe_load_authorities_trusted_by_otp/3 (/app/deps/tls_certificate_check/src/tls_certificate_check_shared_state.erl, line 362)
feature-flag-service  |       in call from tls_certificate_check_shared_state:new_shared_state/2 (/app/deps/tls_certificate_check/src/tls_certificate_check_shared_state.erl, line 335)
feature-flag-service  |       in call from tls_certificate_check_shared_state:handle_shared_state_initialization/2 (/app/deps/tls_certificate_check/src/tls_certificate_check_shared_state.erl, line 262)
feature-flag-service  |       in call from gen_server:try_dispatch/4 (gen_server.erl, line 1123)
feature-flag-service  |       in call from gen_server:handle_msg/6 (gen_server.erl, line 1200)
feature-flag-service  |     ancestors: [tls_certificate_check_sup,<0.1746.0>]
feature-flag-service  |     message_queue_len: 0
feature-flag-service  |     messages: []
feature-flag-service  |     links: [<0.1747.0>]
feature-flag-service  |     dictionary: []
feature-flag-service  |     trap_exit: true
feature-flag-service  |     status: running
feature-flag-service  |     heap_size: 3581853
feature-flag-service  |     stack_size: 28
feature-flag-service  |     reductions: 5621582
feature-flag-service  |   neighbours:
feature-flag-service  | 
feature-flag-service  | =SUPERVISOR REPORT==== 18-Jan-2023::21:41:02.530378 ===
feature-flag-service  |     supervisor: {local,tls_certificate_check_sup}
feature-flag-service  |     errorContext: start_error
feature-flag-service  |     reason: {{badmatch,{error,enoent}},
feature-flag-service  |              [{pubkey_os_cacerts,get,0,
feature-flag-service  |                   [{file,"pubkey_os_cacerts.erl"},{line,38}]},
feature-flag-service  |               {tls_certificate_check_shared_state,
feature-flag-service  |                   maybe_load_authorities_trusted_by_otp,3,
feature-flag-service  |                   [{file,
feature-flag-service  |                        "/app/deps/tls_certificate_check/src/tls_certificate_check_shared_state.erl"},
feature-flag-service  |                    {line,362}]},
feature-flag-service  |               {tls_certificate_check_shared_state,new_shared_state,2,
feature-flag-service  |                   [{file,
feature-flag-service  |                        "/app/deps/tls_certificate_check/src/tls_certificate_check_shared_state.erl"},
feature-flag-service  |                    {line,335}]},
feature-flag-service  |               {tls_certificate_check_shared_state,
feature-flag-service  |                   handle_shared_state_initialization,2,
feature-flag-service  |                   [{file,
feature-flag-service  |                        "/app/deps/tls_certificate_check/src/tls_certificate_check_shared_state.erl"},
feature-flag-service  |                    {line,262}]},
feature-flag-service  |               {gen_server,try_dispatch,4,
feature-flag-service  |                   [{file,"gen_server.erl"},{line,1123}]},
feature-flag-service  |               {gen_server,handle_msg,6,[{file,"gen_server.erl"},{line,1200}]},
feature-flag-service  |               {proc_lib,init_p_do_apply,3,
feature-flag-service  |                   [{file,"proc_lib.erl"},{line,240}]}]}
feature-flag-service  |     offender: [{pid,undefined},
feature-flag-service  |                {id,tls_certificate_check_shared_state},
feature-flag-service  |                {mfargs,{tls_certificate_check_shared_state,start_link,[]}},
feature-flag-service  |                {restart_type,permanent},
feature-flag-service  |                {significant,false},
feature-flag-service  |                {shutdown,infinity},
feature-flag-service  |                {child_type,worker}]
feature-flag-service  | 
feature-flag-service  | =CRASH REPORT==== 18-Jan-2023::21:41:02.573208 ===
feature-flag-service  |   crasher:
feature-flag-service  |     initial call: application_master:init/4
feature-flag-service  |     pid: <0.1745.0>
feature-flag-service  |     registered_name: []
feature-flag-service  |     exception exit: {{shutdown,
feature-flag-service  |                       {failed_to_start_child,
feature-flag-service  |                        tls_certificate_check_shared_state,
feature-flag-service  |                        {{badmatch,{error,enoent}},
feature-flag-service  |                         [{pubkey_os_cacerts,get,0,
feature-flag-service  |                           [{file,"pubkey_os_cacerts.erl"},{line,38}]},
feature-flag-service  |                          {tls_certificate_check_shared_state,
feature-flag-service  |                           maybe_load_authorities_trusted_by_otp,3,
feature-flag-service  |                           [{file,
feature-flag-service  |                             "/app/deps/tls_certificate_check/src/tls_certificate_check_shared_state.erl"},
feature-flag-service  |                            {line,362}]},
feature-flag-service  |                          {tls_certificate_check_shared_state,
feature-flag-service  |                           new_shared_state,2,
feature-flag-service  |                           [{file,
feature-flag-service  |                             "/app/deps/tls_certificate_check/src/tls_certificate_check_shared_state.erl"},
feature-flag-service  |                            {line,335}]},
feature-flag-service  |                          {tls_certificate_check_shared_state,
feature-flag-service  |                           handle_shared_state_initialization,2,
feature-flag-service  |                           [{file,
feature-flag-service  |                             "/app/deps/tls_certificate_check/src/tls_certificate_check_shared_state.erl"},
feature-flag-service  |                            {line,262}]},
feature-flag-service  |                          {gen_server,try_dispatch,4,
feature-flag-service  |                           [{file,"gen_server.erl"},{line,1123}]},
feature-flag-service  |                          {gen_server,handle_msg,6,
feature-flag-service  |                           [{file,"gen_server.erl"},{line,1200}]},
feature-flag-service  |                          {proc_lib,init_p_do_apply,3,
feature-flag-service  |                           [{file,"proc_lib.erl"},{line,240}]}]}}},
feature-flag-service  |                      {tls_certificate_check_app,start,[normal,[]]}}
feature-flag-service  |       in function  application_master:init/4 (application_master.erl, line 142)
feature-flag-service  |     ancestors: [<0.1744.0>]
feature-flag-service  |     message_queue_len: 1
feature-flag-service  |     messages: [{'EXIT',<0.1746.0>,normal}]
feature-flag-service  |     links: [<0.1744.0>,<0.1615.0>]
feature-flag-service  |     dictionary: []
feature-flag-service  |     trap_exit: true
feature-flag-service  |     status: running
feature-flag-service  |     heap_size: 987
feature-flag-service  |     stack_size: 28
feature-flag-service  |     reductions: 232
feature-flag-service  |   neighbours:
feature-flag-service  | 
feature-flag-service  | =INFO REPORT==== 18-Jan-2023::21:41:02.594716 ===
feature-flag-service  |     application: tls_certificate_check
feature-flag-service  |     exited: {{shutdown,
feature-flag-service  |                  {failed_to_start_child,tls_certificate_check_shared_state,
feature-flag-service  |                      {{badmatch,{error,enoent}},
feature-flag-service  |                       [{pubkey_os_cacerts,get,0,
feature-flag-service  |                            [{file,"pubkey_os_cacerts.erl"},{line,38}]},
feature-flag-service  |                        {tls_certificate_check_shared_state,
feature-flag-service  |                            maybe_load_authorities_trusted_by_otp,3,
feature-flag-service  |                            [{file,
feature-flag-service  |                                 "/app/deps/tls_certificate_check/src/tls_certificate_check_shared_state.erl"},
feature-flag-service  |                             {line,362}]},
feature-flag-service  |                        {tls_certificate_check_shared_state,new_shared_state,
feature-flag-service  |                            2,
feature-flag-service  |                            [{file,
feature-flag-service  |                                 "/app/deps/tls_certificate_check/src/tls_certificate_check_shared_state.erl"},
feature-flag-service  |                             {line,335}]},
feature-flag-service  |                        {tls_certificate_check_shared_state,
feature-flag-service  |                            handle_shared_state_initialization,2,
feature-flag-service  |                            [{file,
feature-flag-service  |                                 "/app/deps/tls_certificate_check/src/tls_certificate_check_shared_state.erl"},
feature-flag-service  |                             {line,262}]},
feature-flag-service  |                        {gen_server,try_dispatch,4,
feature-flag-service  |                            [{file,"gen_server.erl"},{line,1123}]},
feature-flag-service  |                        {gen_server,handle_msg,6,
feature-flag-service  |                            [{file,"gen_server.erl"},{line,1200}]},
feature-flag-service  |                        {proc_lib,init_p_do_apply,3,
feature-flag-service  |                            [{file,"proc_lib.erl"},{line,240}]}]}}},
feature-flag-service  |              {tls_certificate_check_app,start,[normal,[]]}}
feature-flag-service  |     type: permanent
feature-flag-service  | 
feature-flag-service  | {"Kernel pid terminated",application_controller,"{application_start_failure,tls_certificate_check,{{shutdown,{failed_to_start_child,tls_certificate_check_shared_state,{{badmatch,{error,enoent}},[{pubkey_os_cacerts,get,0,[{file,\"pubkey_os_cacerts.erl\"},{line,38}]},{tls_certificate_check_shared_state,maybe_load_authorities_trusted_by_otp,3,[{file,\"/app/deps/tls_certificate_check/src/tls_certificate_check_shared_state.erl\"},{line,362}]},{tls_certificate_check_shared_state,new_shared_state,2,[{file,\"/app/deps/tls_certificate_check/src/tls_certificate_check_shared_state.erl\"},{line,335}]},{tls_certificate_check_shared_state,handle_shared_state_initialization,2,[{file,\"/app/deps/tls_certificate_check/src/tls_certificate_check_shared_state.erl\"},{line,262}]},{gen_server,try_dispatch,4,[{file,\"gen_server.erl\"},{line,1123}]},{gen_server,handle_msg,6,[{file,\"gen_server.erl\"},{line,1200}]},{proc_lib,init_p_do_apply,3,[{file,\"proc_lib.erl\"},{line,240}]}]}}},{tls_certificate_check_app,start,[normal,[]]}}}"}
feature-flag-service  | Kernel pid terminated (application_controller) ({application_start_failure,tls_certificate_check,{{shutdown,{failed_to_start_child,tls_certificate_check_shared_state,{{badmatch,{error,enoent}},[{pubkey_os_cacerts,get,0,[{file,"pubkey_os_cacerts.erl"},{line,38}]},{tls_certificate_check_shared_state,maybe_load_authorities_trusted_by_otp,3,[{file,"/app/deps/tls_certificate_check/src/tls_certificate_check_shared_state.erl"},{line,362}]},{tls_certificate_check_shared_state,new_shared_state,2,[{file,"/app/deps/tls_certificate_check/src/tls_certificate_check_shared_state.erl"},{line,335}]},{tls_certificate_check_shared_state,handle_shared_state_initialization,2,[{file,"/app/deps/tls_certificate_check/src/tls_certificate_check_shared_state.erl"},{line,262}]},{gen_server,try_dispatch,4,[{file,"gen_server.erl"},{line,1123}]},{gen_server,handle_msg,6,[{file,"gen_server.erl"},{line,1200}]},{proc_lib,init_p_do_apply,3,[{file,"proc_lib.erl"},{line,240}]}]}}},{tls_certificate_check_app,start,[normal,[]]}}})
feature-flag-service  | 
feature-flag-service  | Crash dump is being written to: erl_crash.dump...done
```

Postgres seems to be running fine:
```
postgres  | The files belonging to this database system will be owned by user "postgres".
postgres  | This user must also own the server process.
postgres  | 
postgres  | The database cluster will be initialized with locale "en_US.utf8".
postgres  | The default database encoding has accordingly been set to "UTF8".
postgres  | The default text search configuration will be set to "english".
postgres  | 
postgres  | Data page checksums are disabled.
postgres  | 
postgres  | fixing permissions on existing directory /var/lib/postgresql/data ... ok
postgres  | creating subdirectories ... ok
postgres  | selecting dynamic shared memory implementation ... posix
postgres  | selecting default max_connections ... 100
postgres  | selecting default shared_buffers ... 128MB
postgres  | selecting default time zone ... Etc/UTC
postgres  | creating configuration files ... ok
postgres  | running bootstrap script ... ok
postgres  | performing post-bootstrap initialization ... ok
postgres  | initdb: warning: enabling "trust" authentication for local connections
postgres  | You can change this by editing pg_hba.conf or using the option -A, or
postgres  | --auth-local and --auth-host, the next time you run initdb.
postgres  | syncing data to disk ... ok
postgres  | 
postgres  | 
postgres  | Success. You can now start the database server using:
postgres  | 
postgres  |     pg_ctl -D /var/lib/postgresql/data -l logfile start
postgres  | 
postgres  | waiting for server to start....2023-01-18 20:27:17.042 UTC [35] LOG:  starting PostgreSQL 14.6 (Debian 14.6-1.pgdg110+1) on x86_64-pc-linux-gnu, compiled by gcc (Debian 10.2.1-6) 10.2.1 20210110, 64-bit
postgres  | 2023-01-18 20:27:17.055 UTC [35] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
postgres  | 2023-01-18 20:27:17.146 UTC [36] LOG:  database system was shut down at 2023-01-18 20:27:16 UTC
postgres  | 2023-01-18 20:27:17.151 UTC [35] LOG:  database system is ready to accept connections
postgres  |  done
postgres  | server started
postgres  | CREATE DATABASE
postgres  | 
postgres  | 
postgres  | /usr/local/bin/docker-entrypoint.sh: ignoring /docker-entrypoint-initdb.d/*
postgres  | 
postgres  | 2023-01-18 20:27:18.259 UTC [35] LOG:  received fast shutdown request
postgres  | waiting for server to shut down....2023-01-18 20:27:18.266 UTC [35] LOG:  aborting any active transactions
postgres  | 2023-01-18 20:27:18.270 UTC [35] LOG:  background worker "logical replication launcher" (PID 42) exited with exit code 1
postgres  | 2023-01-18 20:27:18.280 UTC [37] LOG:  shutting down
postgres  | 2023-01-18 20:27:18.307 UTC [35] LOG:  database system is shut down
postgres  |  done
postgres  | server stopped
postgres  | 
postgres  | PostgreSQL init process complete; ready for start up.
postgres  | 
postgres  | 2023-01-18 20:27:18.410 UTC [1] LOG:  starting PostgreSQL 14.6 (Debian 14.6-1.pgdg110+1) on x86_64-pc-linux-gnu, compiled by gcc (Debian 10.2.1-6) 10.2.1 20210110, 64-bit
postgres  | 2023-01-18 20:27:18.412 UTC [1] LOG:  listening on IPv4 address "0.0.0.0", port 5432
postgres  | 2023-01-18 20:27:18.412 UTC [1] LOG:  listening on IPv6 address "::", port 5432
postgres  | 2023-01-18 20:27:18.415 UTC [1] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
postgres  | 2023-01-18 20:27:18.419 UTC [50] LOG:  database system was shut down at 2023-01-18 20:27:18 UTC
postgres  | 2023-01-18 20:27:18.426 UTC [1] LOG:  database system is ready to accept connections
postgres  | 2023-01-18 21:04:03.349 UTC [1] LOG:  received fast shutdown request
postgres  | 2023-01-18 21:04:03.350 UTC [1] LOG:  aborting any active transactions
postgres  | 2023-01-18 21:04:03.353 UTC [1] LOG:  background worker "logical replication launcher" (PID 56) exited with exit code 1
postgres  | 2023-01-18 21:04:03.353 UTC [51] LOG:  shutting down
postgres  | 2023-01-18 21:04:03.367 UTC [1] LOG:  database system is shut down
postgres  | 
postgres  | PostgreSQL Database directory appears to contain a database; Skipping initialization
postgres  | 
postgres  | 2023-01-18 21:11:44.281 UTC [1] LOG:  starting PostgreSQL 14.6 (Debian 14.6-1.pgdg110+1) on x86_64-pc-linux-gnu, compiled by gcc (Debian 10.2.1-6) 10.2.1 20210110, 64-bit
postgres  | 2023-01-18 21:11:44.281 UTC [1] LOG:  listening on IPv4 address "0.0.0.0", port 5432
postgres  | 2023-01-18 21:11:44.281 UTC [1] LOG:  listening on IPv6 address "::", port 5432
postgres  | 2023-01-18 21:11:44.286 UTC [1] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
postgres  | 2023-01-18 21:11:44.296 UTC [14] LOG:  database system was shut down at 2023-01-18 21:04:03 UTC
postgres  | 2023-01-18 21:11:44.306 UTC [1] LOG:  database system is ready to accept connections
```

**Comment by tsloughter** [comment]
The error being around `pubkey_os_cacerts` makes me think this is related to the upgrade from OTP-23 to 25.

You could test this theory by setting this env variable back to 23.0 and ELIXIR_VERSION back to 1.13.3: https://github.com/open-telemetry/opentelemetry-demo/commit/5a2c0efa923f18bc1f5ae12bbd1843619b351ece#diff-e0a4bade758b455a6570d8dc5a95fb9a408f3460e017637d5b5e410d677259f7R16

**Comment by austinlparker** [comment]
Ah, yeah, the OTP version shouldn't have been updated. I set it back to 23 or whatever a while ago to fix the M1 issues. We need to revert that specific change.

**Comment by julianocosta89** [comment]
Fixed by #698
